tests/phpunit/CRM/Utils/ArrayTest.php

```diff
@@ -367,4 +367,52 @@ public function testFlatten() {
     $this->assertEquals($flat, $expected);
   }
 
+  public function testFormatArrayKeysTransformsAssociativeArrayToSequential() {
+    $tests = [
+      [
+        'object' => [
+          '5' => '1',
+          '1' => '1',
+          '4' => '1'
+        ],
+        'expectation' => ['5', '1', '4']
+      ],
+      [
+        'object' => ['5', '1', '4'],
+        'expectation' => ['5', '1', '4']
+      ],
+      [
+        'object' => ['0', '1'],
+        'expectation' => ['0', '1']
+      ],
+      [
+        'object' => [
+          '0' => '1',
+          '1' => '1'
+        ],
+        'expectation' => ['0', '1']
+      ],
+      [
+        'object' => 'not an array',
+        'expectation' => 'not an array'
+      ],
+      [
+        'object' => [
+          '5' => '1',
+          '1' => '1',
+          '4' => '1',
+          '1' => '1',
+          '4' => '1',
+          '5' => '1',
+          '1' => '1'
+        ],
+        'expectation' => ['5', '1', '4']
+      ],
+    ];
+
+    foreach ($tests as $test) {
```

Comment:
also to be honest and looking at the purpose of this method, it does not seems to be clear, so if the descprtion of the method is supposed to describie it which is : 

```
Convert array where key(s) holds the actual value and value(s) as 1 into array of actual values
```

then I think the following code should be enough : 

function formatArray(&$array) {
    if (!is_array($array) || empty($array)) {
      return;
    }

    $hasAllValuesAsNumberOne = array_unique(array_values($array)) == ['1'];
    if ($hasAllValuesAsNumberOne) {
        $array = array_keys($array);
    }
}

but it also seems that the method should also support things such as : 

```
// handle (0 => 4), (1 => 1)
```

which I am not quite clear what is this but it seems that the value can be something other than 1

Reply:
Yes, there is a bit of confusion so I have to rewrite the initial solution a bit. Thanks!